project4/3700crawler

```diff
@@ -106,43 +154,42 @@ class WebCrawler:
         # Send the request
         sock = self.connect(host)
         headers = {
-            'cookie': self.cookie,
-            'host': host,
-            'content-length': str(len(body)),
-            'content-type': 'application/x-www-form-urlencoded'
+            'Host': host,
+            'Content-Length': str(len(body)),
+            'Content-Type': 'application/x-www-form-urlencoded',
+            'Cookie': self.cookie
         }
         sock.send(str.encode(make_http('POST', path, headers, body)))
 
-        # Get the response
+        # Return the parsed response
         resp = self.get_response(sock)
-
+        resp = self.parse_response(resp)
         return resp
 
     def login(self):
         """Logs into Fakebook"""
+        # Get the log-in form
         login_form = self.GET(LOGIN_FORM)
-        match = re.search('[Ss]et-[Cc]ookie:\W*csrftoken=(\w+);', login_form)
-        if (not match):
-            print('Couldn\'t extract CSRF token, aborting')
-            print(login_form)
-            exit(1)
-        self.csrf = match.group(1)
-        match = re.search('[Ss]et-[Cc]ookie:\W*sessionid=(\w+);', login_form)
-        if (not match):
-            print('Couldn\'t extract SessionID token, aborting')
-            print(login_form)
-            exit(1)
-        self.session = match.group(1)
+        
+        # Loop until the form has been successfully retrieved
+        while login_form['Status Code'] != str(200):
```

Comment:
Don't like that this can be an infinite loop, I don't know if the project said that the random 400's apply to the login page but since it'll fail immediately if that's the case and we're starting this tool by hand, I'd rather not do this